src/Agda2Hs/Compile/Record.hs

```diff
@@ -168,7 +169,7 @@ compileRecord target def = setCurrentRange (nameBindingSite $ qnameName $ defNam
       return $ Hs.DataDecl () don Nothing hd [conDecl] ds
 
 checkUnboxPragma :: Defn -> C ()
-checkUnboxPragma def@Record{ recFields = (f:fs) }
-  | keepArg f , all (not . keepArg) fs , not (recRecursive def) = return ()
-checkUnboxPragma _ = genericError $
-  "An unboxed type must be a non-recursive record type with exactly one non-erased field."
+checkUnboxPragma def@Record{recFields} =
+  if length (filter keepArg recFields) == 1 && not (recRecursive def)
+    then return ()
+    else genericError "An unboxed type must be a non-recursive record type with exactly one non-erased field."
```

Comment:
This will raise a `Non-exhaustive pattern matching` error now that you've removed the catch all clause.
The best option IMO is:
```hs
checkUnboxPragma def
  | Record{recFields} <- def
  , length (filter keepArg recFields) == 1
  , not (recRecursive def)
  = return ()
  | otherwise
  = genericError "An unboxed type must be a non-recursive record type with exactly one non-erased field."
```

Reply:
Oops, indeed, let's keep the case!